charts/kyverno/templates/networkpolicy.yaml

```diff
@@ -30,4 +30,77 @@ spec:
   ingress:
     - {}
   {{- end }}
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-prometheus
+  namespace: kyverno
+  labels:
+    app: kyverno
+spec:
+  podSelector:
+    matchLabels:
+      app: kyverno
+  ingress:
+  - from:
+    {{- with .Values.networkPolicy.prometheusNamespaceLabels }}
+    - namespaceSelector:
+        matchLabels:
+            {{ toYaml . | nindent 10 }}
+    {{- end }}
+    {{- with .Values.networkPolicy.prometheusPodLabels }}
+    - podSelector:
+        matchLabels:
+            {{ toYaml . | nindent 10 }}
+    {{- end }}
+    ports:
+    # Webhook access
+    - protocol: TCP
+      port: 9443
+    # Allow prometheus scrape for metrics
+    - protocol: TCP
+      port: 8000
+  policyTypes:
+    - Ingress
+    - Egress
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-apiserver
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+  - Egress
+  # Allow ingress from kube-apiserver on port 9443
+  ingress:
+    - from:
+        - namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: kube-system
+        - podSelector:
+            matchLabels:
+              component: kube-apiserver
+              tier: control-plane
+      ports:
+        - port: 9443
+  # If openEgress is true allow all egress traffic else allow egress to only kube-apiserver on port 9443
+  {{- if .Values.networkPolicy.openEgress }}
+  egress:
+  - {}
+  {{- else }}
+  egress:
+    - to:
+        - namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: kube-system
+        - podSelector:
+            matchLabels:
+              component: kube-apiserver
+              tier: control-plane
```

Comment:
The `namespaceSelector` and `podSelector` need to be in same array item, otherwise you are allowing everything from `kube-system` or `kube-apiserver` from same namespace as Kyverno